var/spack/repos/builtin/packages/tau/package.py

```diff
@@ -146,3 +146,9 @@ def link_tau_arch_dirs(self):
                 dest = join_path(self.prefix, d)
                 if os.path.isdir(src) and not os.path.exists(dest):
                     os.symlink(join_path(subdir, d), dest)
+
+    def setup_environment(self, spack_env, run_env):
+        pattern = '%s/Makefile.*' % self.prefix.lib
+        files = glob.glob(pattern)
+        makefile = files[0] if files else ''
+        run_env.set('TAU_MAKEFILE', makefile)
```

Comment:
What happens if `TAU_MAKEFILE` is set to `''`?  If this is an error, then why the error checking here?  Why not just: `run_env.set('TAU_MAKEFILE', files[0])`?